machine-learning/README.md

```diff
@@ -1,4 +1,99 @@
+
+There are a lot more! Browse [devpost](https://devpost.com) to see more cool projects. 
+
+## Tools and libraries to get you started
+If you're gonna do any kind of ML project, I'd highly recommend Python. Many common Python ML libraries such as Keras, PyTorch, and Tensorflow abstract alot of the process for you, so you can focus on the bigger picture! This is especially good for beginners who may not understand everything that is going on behind the scenes just yet. C++ and Java are also pretty popular options, but I can't speak to their usability.
+
+Keras, PyTorch, and Tensorflow have a whole bunch of documentation out there for how to set them up, and how to use them. I'll link a couple python tutorials I found really useful down below. The best part about this library is that it works right out of the box (for the most part), allowing you to quickly move right on integrating it with the rest of the project.
+
+You will most likely have to look into these frameworks to decide which one is the right fit for your project, but here are some general differences as a starting point:
+
+Keras (Recommended for beginners and small datasets):
+  - High level API (Abstracts things more)
+  - Slower runtimes(Shouldn't matter too much in this context)
+  - Simplest to use
+  - Least lines of code
+  - Less accurate models
+  - Simple models with minial debugging  
+  
+Tensorflow (Recommended for experienced users and high performance reduced size models):
+  - High and Low level API's
+  - Faster runtimes
+  - Harder to use
+  - Reduced code needed
+  - Accurate models
+  - Hard to debug  
+  
+ PyTorch (Recommended for machine learning gods and high performance models with large datasets):
+  - Low Level API
+  - Faster runtimes
+  - Hardest to use
+  - Most lines of code needed
+  - Extremely accurate models
+  - Good debugging support
+  
+
+## Pros and Cons
+While ML is indeed very flashy, it also has some not-so-flashy aspects, which I will cover so you can consider if incorporating ML into your project is a good fit  
+Pros:
+  - Very relevant in the field of software development
+  - Very flashy
+  - Very powerful libraries
+  - Generally beginner friendly
+  - Applicable to many projects  
+  
+Cons:
+  - Long installation (You'll probably spend more time installing dependancies and libraries than actually working on your project)
+  - Needs large dataset (Machine Learning generally needs extremely large datasets to be accurate. In most cases, I have found myself       without enough resources or enough time to create a custom dataset. And it is usually disappointing when it doesn't correclt           predict in front of the judges)
+  - Reading thick documentation (Because of how powerful these libraries are, you will need to take some time to figure out which tools     are the ones you need and which ones you don't)
```

Comment:
```suggestion
  - Reading thick documentation (Because of how powerful these libraries are, you will need to take some time to figure out which tools are the ones you need and which ones you don't)
```